### PR TITLE
VACMS-9358: add audit opening hours comments field script

### LINKED ISSUE #9358: Audit Opening Hours Comments fields
## User Story or Problem Statement

As a designer or engineer working on Opening Hours, i want to know what editors have added in comment fields

In #8935 we removed time zone content from Opening Hours comment fields. What's left? This will help us learn how editors are using Comments fields, so that we can 
1. Provide better guidance about what to put in Comments
2. Make improvements to the Hours widget, such as a "Closed" option

## References
The following scripts may have some value in terms of code reusability:
1. Script `VACMS-6931-switch-service-locations-office-hours.php` writes to a CSV file
2. Script `VACMS-8935-scrub-tz-in-hours-field-comment.php` manipulates Opening Hours data

## Field Sources
The "Office Hours" field is present on the following content types:
- VAMC Facility (health_care_local_facility - field_office_hours)
- VAMC System Billing and Insurance (vamc_system_billing_insurance - field_hours_for_copay_inquiries_)
- Vet Center (vet_center - field_office_hours)
- Vet Center - Community Access Point (vet_center_cap - field_office_hours)
- Vet Center - Mobile Vet Center (vet_center_mobile_vet_center - field_office_hours)
- Vet Center - Outstation (vet_center_outstation - field_office_hours)

The "Office Hours" field is also present on the following paragraph types:
- Service Location (service_location)

The following content types use the Service Location paragraph type:
- VAMC Facility Health Service (health_care_local_health_service)
- VAMC Facility Non-clinical Service (vha_facility_nonclinical_service)

## Acceptance Criteria

- [ ] We have script(s) to generate CSV(s) showing any Opening Hours data that has comments
- [ ] Should include fields: node id, content type, url, moderation state, comment
- [ ] We may need two CSVs, one for Service Location data. Ideally, it would also show the content type and NID that references it. 
- [ ] Scripts are documented somewhere.

<details><summary>Old ones</summary>

- [ ] display machine, not `default`
- [ ] URL [`/admin/content/audit/opening-hours view`]
- [ ] Menu item setting: parent audit, label "Opening hours"
- [ ] limit to 100 per page, paginated
- [ ] show result count in header
- [ ] default columns
  - [ ] Title (sortable) 
  - [ ] Edit button (not operations, no column heading) 
  - [ ] Content type (sortable)
  - [ ] Moderation State
  - [ ] Section (field_administration) (sortable)
  - [ ] Is Locked (optional)
  - [ ] Last updated (date time and by user) (sortable, default sort, DESC)
  - [ ] Fields (combine into one header)
    - [ ] field_hours_for_copay_inquiries_
   -  [ ] field_office_hours
- [ ] filters
  - [ ] text search (exposed)
  - [ ] content type (exposed): 
    - [ ] multiselect with the following options:  VAMC System Billing and Insurance, Vet Center - Community Access Point, VAMC Facility, Vet Center, Vet Center - Mobile Vet Center Vet Center - Outstation. 
    - [ ] don't allow other content types
    - [ ] preseslect the following: VAMC System Billing and Insurance Vet Center - Community Access Point Service location
  - [ ] moderation state (exposed)
  - [ ] section (exposed) (workbench access section; single select, show children)
  - [ ] "Copay inquriries hours": `field_hours_for_copay_inquiries_`: is any / is empty / is not empty 
  - [ ] "Hours": `field_office_hours`: is any / is empty / is not empty 
  - [ ] Reset button exists
  - [ ] Apply button says 'Filter'

Stretch
- Someway of filtering based on whether Comments are filled out. 

</details>

## Implementation steps

Updating scripts from #6931 and  #8935

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [x] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [x] `⭐️ Facilities`
- [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #9358 

## Testing done

Visual / Manual testing

## Screenshots

Screen Shot 2022-06-08 at 6 26 39 PM

## QA steps

From a terminal:
1. Run the command `drush scr ./scripts/content/VACMS-8935-scrub-tz-in-hours-field-comment.php`
2. Check the docroot for the project - verify a new file was created: `docroot/office_hours_comment_audit.csv`

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [x] `⭐️ Facilities`
- [ ] `⭐️ User support`

### CODE REVIEW

**Comment by JayDarnell** [comment]
A sample CSV created form my local machine:
[office_hours_comment_audit.csv](https://github.com/department-of-veterans-affairs/va.gov-cms/files/8865403/office_hours_comment_audit.csv)

**Comment by kevwalsh** [comment]
I've done some auditing now, [in this airtable](https://airtable.com/appxsEaJnhCDMPeD2/tblm7ZLbTl7GIeiZG/viwXpy3X1DwrUTYWs?blocks=show). This is awesome. 

Does it only report unique comments per field? So of M-F say "By appointments only" and Sat/Sun are "Closed", will it only report two instances instead of 7? 

Just want to make sure that was intentional. 

I have no idea if this needs merging, hard to say how often we will need this.

**Comment by kevwalsh** [comment]
Here's what i was able to do with this data. 

Opening_hours___Operating_hours_comments__Hours_dat_-_Airtable

File: scripts/content/VACMS-9358-audit-opening-hours-comments.php [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    ->log(LogLevel::INFO, 'Nodes: %current nodes hours comments audited. Nodes processed: %nids', [
+      '%current' => $sandbox['node_current'],
+      '%nids' => implode(', ', $nids),
+    ]);
+
+  $sandbox['#nodes_finished'] = ($sandbox['node_current'] / $sandbox['node_total']);
+
+  // Log the all-finished notice.
+  if ($sandbox['#nodes_finished'] == 1) {
+    Drupal::logger('va_gov_db')->log(LogLevel::INFO, 'RE-saving all %count nodes completed by va_gov_force_save_remove_timezones_in_comments', [
```
Comment (swirtSJW) [inline scripts/content/VACMS-9358-audit-opening-hours-comments.php:124]: Resaving?  I thought this was just an audit?

File: scripts/content/VACMS-9358-audit-opening-hours-comments.php [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      '%count' => $sandbox['node_total'],
+    ]);
+    return "Node audit complete. {$sandbox['node_current']} / {$sandbox['node_total']}\n";
+  }
+
+  return "Processed nodes... {$sandbox['node_current']} / {$sandbox['node_total']}.\n";
```
Comment (swirtSJW) [inline scripts/content/VACMS-9358-audit-opening-hours-comments.php:130]:
```suggestion
  return "Audited nodes... {$sandbox['node_current']} / {$sandbox['node_total']}.\n";
```

File: scripts/content/VACMS-9358-audit-opening-hours-comments.php (hunk 1 above)
Comment (swirtSJW) [inline scripts/content/VACMS-9358-audit-opening-hours-comments.php:124]:
```suggestion
    Drupal::logger('va_gov_db')->log(LogLevel::INFO, 'Auditing all %count nodes completed by VACMS-9358-audit-opening-hours-comments', [
```

**Review by swirtSJW — COMMENTED** [review]
Looks good overall.  Just some misleading messaging.

**Review by ethanteague — APPROVED** [review]
:+1:

**Pull request merged**